src/pyedb/dotnet/edb_core/edb_data/layer_data.py

```diff
@@ -238,6 +238,26 @@ def type(self, value):
         self._type = value
         self._pedb.stackup._set_layout_stackup(layer_clone, "change_attribute")
 
+    @property
+    def properties(self):
+        data = {}
+        data["name"] = self.name
+        data["type"] = self.type
+        data["color"] = self.color
+        return data
+
+    @properties.setter
+    def properties(self, params):
+        name = params.get("name")
+        if name:
+            self.name = name
+        type = params.get("type")
+        if type:
+            self.type = type
+        color = params.get("color")
+        if color:
+            self.color = color
```

Comment:
```suggestion
        name = params.get("name", "")
        if name:
            self.name = name
        type = params.get("type", "")
        if type:
            self.type = type
        color = params.get("color", "")
        if color:
            self.color = color
```
Adding default value will ensure that if someone does not provide one of the expected parameters, this call will not raise an exception. However, if you want an exception to be triggered because you want users to provide parameters, discard my changes.